src/CADL.Extension/Emitter.Csharp/src/emitter.ts

```diff
@@ -189,6 +193,41 @@ function createModel(program: Program): any {
     }
 }
 
+function processServiceAuthentication(
+    authentication: ServiceAuthentication
+): InputAuth {
+    const auth = {} as InputAuth;
+    for (const option of authentication.options) {
+        for (const schema of option.schemes) {
+            switch (schema.type) {
+                case "apiKey":
+                    auth.ApiKey = schema.name;
+                    break;
+                case "oauth2":
+                    let scopes: string[] = [];
```

Comment:
We probably need `let scopes = new Set<string>()` here to avoid repeated scopes.

Reply:
Thanks @AlexanderSher Good suggestion. I will update.